### PR TITLE
chore: keep typedArray signature consistent

### PR DESCRIPTION
**Pull request by yjhmelody**
I found that the current signature definition of the typeArray interface is not strictly consistent.

There may be some discrepancies with the documentation website.

### CODE REVIEW

File: std/assembly/index.d.ts [hunk 1]
```diff
@@ -1581,9 +1581,9 @@ declare abstract class TypedArray<T> implements ArrayBufferView {
   /** The lastIndexOf() method returns the last index at which a given element can be found in the typed array, or -1 if it is not present. The typed array is searched backwards, starting at fromIndex. */
   lastIndexOf(searchElement: T, fromIndex?: i32): i32;
   /** Returns copied section of an TypedArray from begin inclusive to end exclusive */
-  slice(begin?: i32, end?: i32): TypedArray<T>;
+  slice(begin?: i32, end?: i32): this;
```
Comment (MaxGraey) [inline std/assembly/index.d.ts:1584]: The "this" type should be returned only when the body of the function contains "return this", that is, it returns a reference to the current instance. This is not the case with "slice" or "subarray". What is returned is a reference to a new instance

Comment (yjhmelody) [reply std/assembly/index.d.ts:1584]: As I understand it, this only represents the type, and there is no semantic requirement that the value of this must be returned (it is also not guaranteed).

Comment (MaxGraey) [reply std/assembly/index.d.ts:1584]: I propose following semantic convention which reflect is it "new instance" (TypedArray<T>) or "the same instance" (this)

Comment (MaxGraey) [reply std/assembly/index.d.ts:1584]: Just see how TS automatically infer return type in playground:
https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8CwAoa0CWA7AZgKYBOAghACoAW2EAFAJQoZZbGEAuArsbtBzQgtoAXwzC8RMhACSuCBzC5ghRs0yt23XtFyEA7nETCx6EUA

Comment (yjhmelody) [reply std/assembly/index.d.ts:1584]: Ok you convinced me

**Review by MaxGraey — APPROVED** [review]

File: std/assembly/typedarray.ts [hunk 2]
```diff
@@ -94,27 +94,27 @@ export class Int8Array extends ArrayBufferView {
     return REDUCE_RIGHT<Int8Array, i8, T>(this, fn, initialValue);
   }
 
-  map(fn: (value: i8, index: i32, self: Int8Array) => i8): Int8Array {
+  map(fn: (value: i8, index: i32, self: this) => i8): Int8Array {
```
Comment (dcodeIO) [inline std/assembly/typedarray.ts:97]: In AS, the `this` type is static, so here it is always `Int8Array` regardless of subtyping. As such, with this change, TS would infer something different than AS when the method is overloaded, which is probably not what you want.

Comment (yjhmelody) [reply std/assembly/typedarray.ts:97]: Ok, Maybe I should close this PR :(.

**Pull request closed without merging**